[DSS-462] add Popover (#100)
Kajabi/sage: libs/core/src/components/pds-popover/stories/pds-popover.docs.mdx

import { Meta, Canvas, ArgTypes } from '@storybook/blocks';

import * as stories from './pds-popover.stories.js';

<Meta of={stories}
/>

# Popover

Popovers are containers that float on top of existing content used to display additional content.

## Properties

<ArgTypes of={stories} />

## Variants

### Trigger
Whether your trigger is a `pds-button` or a different component or HTML element, it will be placed in the default slot between the opening and closing tags.
There can only be one trigger for the Popover.

#### Button trigger

<Canvas mdxSource={`
<pds-popover placement="bottom-right" html-content="true">
  <div slot="content">
    <p><strong>This is a popover</strong></p>
  </div>
  <pds-button variant="secondary">Click</pds-button>
</pds-popover>
`}>

>   <pds-popover placement="bottom-start" html-content="true">

Reviewer: Just a thought, but since we now display our own source code separately, I think we could add something like `<div style={{height: "200px"}}>` to the examples here to show the entire popover and it will not display it in the code example. Currently, all the examples get cut off.

<img width="577" alt="Screenshot 2023-09-12 at 3 36 47 PM" src="https://github.com/Kajabi/pine/assets/1175111/8af3c90b-d6b1-4890-8d23-534a9410c39a">

import { Meta, Canvas, ArgTypes } from '@storybook/blocks';

import * as stories from './pds-popover.stories.js';

<Meta of={stories}
/>

# Popover

> Popovers are containers that float on top of existing content used to display additional content.

Reviewer 2: ```suggestion
Popover is a discrete UI element that displays over the main content to present extra information or user options. It's designed for minimal disruption, appearing only when necessary and allowing for a clutter-free, user-centric experience.```

## Properties

<ArgTypes of={stories} />

## Variants

### Trigger

> Whether your trigger is a `pds-button` or a different component or HTML element, it will be placed in the default slot between the opening and closing tags.

Reviewer 2: ```suggestion
The trigger can be a button, `pds-button`, another Pine component, or an HTML element. There is just one trigger for each Popover, making its use straightforward.
```

    <div>
      <pds-button variant="secondary">Cancel</pds-button>
      <pds-button>Get Started</pds-button>
    </div>
  </div>
  <pds-button variant="secondary">Popover trigger</pds-button>
</pds-popover>
`}>
  <div style={{ height: '400px' }}>
    <pds-popover placement="bottom-start">
      <div slot="content">
        <p>Pastrami chuck leberkas, swine biltong tail fatback jowl landjaeger ground round strip steak t-bone cow beef. Filet mignon bresaola pastrami, beef salami hamburger short ribs. Strip steak picanha pork loin tri-tip, prosciutto frankfurter turducken cupim sausage corned beef spare ribs sirloin fatback t-bone bacon. Boudin pork chop capicola short ribs flank rump. Fatback andouille chuck, pork venison t-bone bacon cupim turducken brisket sausage pork loin. Shank buffalo sirloin swine shoulder sausage prosciutto fatback.</p>
        <div>
          <pds-button variant="secondary">Cancel</pds-button>
          <pds-button>Get Started</pds-button>
        </div>
      </div>
      <pds-button variant="secondary">Popover trigger</pds-button>
    </pds-popover>
  </div>
</Canvas>

### Positioning

> Using the placement prop you can adjust where your tooltip will be displayed.
> 

Reviewer 2: ```suggestion
The placement property enables precise control over the positioning of your Popover. By utilizing this property, you can dictate the specific location of the popover relative to the trigger element. This customization ensures optimal visibility and alignment per your page's layout and design requirements.
```

### Arrow

> By default the arrow is hidden on popovers. It can be shown by enabling it when `has-arrow` is `true`.

Reviewer 2: ```suggestion
By default the arrow is hidden. It can be shown by setting `has-arrow` to `true`.
```
